WP_Auth0.php

```diff
@@ -436,27 +436,50 @@ public static function uninstall() {
 		delete_transient( WPA0_JWKS_CACHE_TRANSIENT_NAME );
 	}
 
+	/**
+	 * Look for a class within a specific set of paths.
+	 *
+	 * @param string $class - Class name to look for.
+	 *
+	 * @return bool
+	 */
 	private function autoloader( $class ) {
-		$path = WPA0_PLUGIN_DIR;
-		$paths = array();
-		$exts = array( '.php', '.class.php' );
-
-		$paths[] = $path.'lib/';
-		$paths[] = $path.'lib/admin/';
-		$paths[] = $path.'lib/exceptions/';
-		$paths[] = $path.'lib/wizard/';
-		$paths[] = $path.'lib/initial-setup/';
-		$paths[] = $path.'lib/twitter-api-php/';
-		$paths[] = $path.'lib/php-jwt/Exceptions/';
-		$paths[] = $path.'lib/php-jwt/Authentication/';
-		$paths[] = $path;
-
-		foreach ( $paths as $p ) {
-			foreach ( $exts as $ext ) {
-				if ( file_exists( $p.$class.$ext ) ) {
-					require_once $p.$class.$ext;
-					return true;
-				}
+		$source_dir = WPA0_PLUGIN_DIR . 'lib/';
+
+		// Catch non-name-spaced classes that still need auto-loading.
+		switch ( $class ) {
+			case 'TwitterAPIExchange':
+				require_once $source_dir . 'twitter-api-php/' . $class . '.php';
+				return true;
+
+			case 'JWT':
+				require_once $source_dir . 'php-jwt/Authentication/' . $class . '.php';
+				return true;
+
+			case 'BeforeValidException':
+			case 'ExpiredException':
+			case 'SignatureInvalidException':
+				require_once $source_dir . 'php-jwt/Exceptions/'  . $class . '.php';
+				return true;
+		}
+
+		// Anything that's not part of the above and not name-spaced can be skipped.
+		if ( 0 !== strpos( $class, 'WP_Auth0' ) ) {
+			return false;
+		}
+
+		$paths = array(
+			$source_dir,
+			$source_dir . 'admin/',
+			$source_dir . 'exceptions/',
+			$source_dir . 'wizard/',
+			$source_dir . 'initial-setup/',
+		);
+
+		foreach ( $paths as $path ) {
```

Comment:
I don't know how often this method would be called, but I suppose several times per execution. Is there any way of iterating through all the files in a more efficient way? Maybe accepting an array of "classes to require" so you can iterate once and require them all.

Reply:
> several times per execution

Correct

> Maybe accepting an array of "classes to require"

That would defeat the purpose of the autoloader. This is here to "magically" include the necessary files for whatever class you might call. There is a more efficient way to do this using [Namespaces](https://secure.php.net/manual/en/language.namespaces.php) but that would require a lot more work (I have it on my list)